src/service/api.tsx

```diff
@@ -67,13 +75,19 @@ const api = {
 
   setHistoryId(id: number) {
     const store = this.getStore();
-    const isThereAnId = store.history.includes(id);
-    if (isThereAnId) {
-      store.history.push(id)
-    };
+    const isNewId = store.history.includes(id);
+    if (!isNewId) {
+      store.history.push(id);
```

Comment:
It seems like here can be used approach from comment of @oleksandrstarov too (!store.favorites.includes(id))

Reply:
fixed